File: TourismAgency/AgencyDomain/src/main/java/domain/Trip.java
package domain;

import javax.persistence.Basic;
import javax.persistence.Column;
import javax.persistence.Table;
import java.time.LocalDateTime;
import java.time.LocalTime;

@javax.persistence.Entity
@Table(name = "trips")
public class Trip extends Entity{
    @Basic
    @Column(name = "place")
    private String place;
    @Basic
    @Column(name = "transport")
    private String transport;
    @Basic
    @Column(name = "dateTrip")
    private LocalDateTime date;

    public Trip() {
    }

    @Basic
    @Column(name = "price")
    private Double price;
    @Basic
    @Column(name = "nrTickets")
    private Integer nrTickets;
    @Basic
    @Column(name = "freeTickets")
    private Integer freeTickets;

    public Trip(String place, String transport, LocalDateTime date, Double price, Integer nrTickets, Integer freeTickets) {
        this.place = place;
        this.transport = transport;
        this.date = date;
        this.price = price;
        this.nrTickets = nrTickets;
        this.freeTickets = freeTickets;
    }

    public Trip(Long tripID,String place,Integer freeTickets) {
        super();
        this.setId(tripID);
        this.place = place;
        this.transport = "";
        this.date = LocalDateTime.MIN;
        this.price = 0d;
        this.nrTickets = 0;
        this.freeTickets = freeTickets;
    }

    public String getPlace() {
        return place;
    }

    public void setPlace(String place) {
        this.place = place;
    }

    public String getTransport() {
        return transport;
    }

    public void setTransport(String transport) {
        this.transport = transport;
    }

    public LocalDateTime getDate() {
        return date;
    }
    public LocalTime getDateTime(){return date.toLocalTime();}

    public void setDate(LocalDateTime date) {
        this.date = date;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public Integer getNrTickets() {
        return nrTickets;
    }

    public void setNrTickets(Integer nrTickets) {
        this.nrTickets = nrTickets;
    }

    public Integer getFreeTickets() {
        return freeTickets;
    }

    public void setFreeTickets(Integer freeTickets) {
        this.freeTickets = freeTickets;
    }

    @Override
    public String toString() {
        return "Trip{" +
                "id"+getId()+
                ",  place='" + place + '\'' +
                ", transport='" + transport + '\'' +
                ", date=" + date +
                ", price=" + price +
                ", nrTickets=" + nrTickets +
                ", freeTickets=" + freeTickets +
                '}';
    }
}

File: AgentieTurism/model/domain/Trip.cs
using System;
using Newtonsoft.Json;

namespace model.domain
{
    [Serializable]
    public class Trip :Entity<long>
    {
        

        [JsonProperty("place")]
        public string Place { get; set; }
        [JsonProperty("transport")]
        public string Transport { get; set; }
        [JsonProperty("date")]
        public string Date { get; set; }
        [JsonProperty ("price")]
        public double Price { get; set; }
        [JsonProperty ("nrTickets")]
        public int NrTickets { get; set; }
        [JsonProperty ("freeTickets")]
        public int FreeTickets { get; set; }

        public Trip() { }
       
        public Trip(string place, string transport, string data, double price, int tickets, int freeTickets)
        {
            this.Place = place;
            this.Transport = transport;
            Date = data;
            this.Price = price;
            NrTickets = tickets;
            this.FreeTickets = freeTickets;
        }

        public Trip(string place, string transport, string data, double price, int tickets)
        {
            this.Place = place;
            this.Transport = transport;
            Date = data;
            this.Price = price;
            NrTickets = tickets;
            FreeTickets = tickets;
        }

        public Trip(long tripId, string place, int avTickets)
        {
            ID = tripId;
            Transport = "";
            Date = DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss");
            Price = 0;
            NrTickets = avTickets;
            this.Place = place;
            FreeTickets = avTickets;
        }

        public override string ToString()
        {
            return string.Format("[Trip: id={0}, place={1}, transport={2}, date={3}, price={4}, nrTickets={5}, freeTickets={6}]", ID, Place, Transport,Date,Price,NrTickets,FreeTickets);
        }
    }
}
